### PR TITLE
GOD-137 fixups top meet jsonapi spec, including links in includes

### PR DESCRIPTION
**Pull request by JliptonRTR**

### CODE REVIEW

**Comment by JliptonRTR** [comment]
I'm going to change the names, just wanted to get this out there to help unblock john

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java [hunk 1]
```diff
@@ -133,8 +134,22 @@ private ResourceObjectImpl buildEntity(final Mapping mapping, final Object entit
...
+			if ("undefined".equals(type)) {
+				type = model.type();
+			}
+			if ("undefined".equals(type)) {
+				return null;
+			}
+			return type;
+		}
+		return null;
+	}
```
Comment (jadeglaze) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/filter/JsonApiResponseFilter.java]: duplicate code?

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/JSONAPI.java [hunk 2]
```diff
@@ -83,7 +88,69 @@ public ApiDocument build(UriInfo uriInfo) {
...
+			if (!links.isEmpty()) {
+				doc.setLinks(links);
+			}
+			if (meta != null) {
+				doc.setMeta(meta);
+			}
+			return doc;
+		}
+
+		private String getType(D data) {
```
Comment (jadeglaze) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/JSONAPI.java]: don't like the name getType. Sounds like a typical getter... not too big a deal though.

**Comment by jadeglaze** [comment]
lgtm

File: dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java [hunk 3]
```diff
@@ -1,6 +1,7 @@
 package io.rtr.jsonapi.util;
 
 import com.google.common.collect.Lists;
+import io.rtr.jsonapi.annotation.ApiModel;
```
Comment (zain-cheng) [inline dropwizard-jsonapi/src/main/java/io/rtr/jsonapi/util/FieldUtil.java]: unnecessary

**Comment by zain-cheng** [comment]
lgtm

**Pull request merged**